algorithms/src/snark/varuna/varuna.rs

```diff
@@ -437,15 +433,13 @@ where
             &mut sponge,
         );
 
-        let (verifier_third_msg, verifier_state) =
-            AHPForR1CS::<_, MM>::verifier_third_round(verifier_state, &mut sponge)?;
+        prover_state.verifier_state.as_mut().unwrap().third_round(&mut sponge)?;
```

Comment:
Maybe we can add `ProverState::verifier_nth_round` functions which handle these things?`?

Reply:
c2ea3e552